tendrl/monitoring_integration/sync/dashbaord_sync.py

```diff
@@ -1,155 +1,178 @@
+                        self.create_dashboard(
                             dashboard_name,
-                            cluster_detail_list
+                            cluster_details
                         )
+
                 else:
-                    self.create_all_dashboard(
+                    self.create_dashboard(
                         dashboard_name,
-                        cluster_detail_list
+                        cluster_details
                     )
 
-    def create_all_dashboard(self, dashboard_name, cluster_detail_list):
-        if dashboard_name == "nodes":
-            dashboard_name = "hosts"
-        try:
-            create_alert_dashboard.CreateAlertDashboard(
-                dashboard_name,
-                cluster_detail_list
-            )
-        except (etcd.EtcdKeyNotFound, KeyError) as error:
-            logger.log(
-                "error",
-                NS.get("publisher_id", None),
-                {
-                    'message': "Failed to create dashboard"
-                    "with error {0}".format(str(error))
-                }
-            )
+    def create_dashboard(self, resource_name, cluster_details):
+        flag = False
+        for resource in cluster_details[resource_name]:
+            if not flag:
+                # create dashboard
+                create_alert_dashboard.create_resource_dashboard(
+                    resource_name,
+                    resource,
+                    cluster_details["sds_name"],
+                    cluster_details["integration_id"]
+                )
+                flag = True
+            else:
+                # update dashboard
+                self.create_gluster_resource(
+                    resource,
+                    resource_name,
+                    cluster_details["integration_id"],
+                    cluster_details["sds_name"]
+                )
 
-    def create_resource(self, cluster_detail_list, resource_type):
-        update_dashboard = UpdateDashboard()
-        self.map = {"cluster": "at-a-glance", "host": "nodes",
-                    "volume": "volumes", "brick": "bricks"}
-        for cluster in cluster_detail_list:
-            integration_id = cluster.integration_id
-            if resource_type == "volumes":
-                for volume in cluster.volumes:
-                    try:
-                        resource_name = str(volume["name"])
-                        response = update_dashboard._add_panel(
-                            integration_id,
-                            self.map["volume"],
-                            resource_name
-                        )
+    def create_gluster_resource(self, resource, resource_type,
+                                integration_id, sds_name):
+        if resource_type == "volumes":
+            for volume in resource:
+                try:
+                    resource_name = str(volume["name"])
+                    response = create_alert_dashboard.add_panel(
```

Comment:
How these three calls for create_alert_dashboard.add_panel different? Only resource_name forming is different. I feel this whole thing could be written as below

```
if resoource_type == "bricks":
    resource_name = <some logic to form the name>
elif Resource_type == "volumes":
    resource_name = <some logic to form the name>
elif resource_type == "hosts":
    resource_name = <some logic to form the name>
try:
    response = create_alert_dashboard(integration_id, resource_type, resource_name, sds_name)
except:
    <handle>
```

Reply:
i will modify it